Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -134,15 +157,43 @@ driverEndRide handle rideId req = do
     $ DriverReq req
 
 callBasedEndRide ::
-  (MonadThrow m, Log m, MonadTime m, MonadGuid m, CoreMetrics m, MonadReader r m, HasField "enableAPILatencyLogging" r Bool, HasField "enableAPIPrometheusMetricLogging" r Bool) =>
+  ( MonadThrow m,
+    Log m,
+    MonadTime m,
+    MonadGuid m,
+    CoreMetrics m,
+    MonadReader r m,
+    HasField "enableAPILatencyLogging" r Bool,
+    HasField "enableAPIPrometheusMetricLogging" r Bool,
+    HasFlowEnv m r '["smsCfg" ::: SmsConfig],
+    HasCacheConfig r,
+    EsqDBFlow m r,
+    EsqLocDBFlow m r,
+    EncFlow m r,
+    Redis.HedisFlow m r
+  ) =>
```

Comment:
the same

Reply:
I suggest to add synonym type `EndRideFlow m r = ...`
because these constraints are too huge and duplicated

Reply:
By the way can we add new function in `ServiceHandle` instead of changing constraint?